### PR TITLE
feat(builder): Apply standard theming for shadCn and centralize providers

### PR DESCRIPTION
**Pull request by andrewhooker2**
**User description**
### Background
The current implementation of the ShadCn components is in need of a theme standardization. When developing front-end solutions it is essential that we stay in alignment on styles, themes, and ux. 

By implementing this change we can allow developers to use class names such as "bg-background", "bg-accent", etc. This will remove complexity and drive easy development of new additions to the solution. 

The only other change was a consolidation of providers. This will keep our root layout clean and robust. Allowing for the focus of the layout.tsx to be components and not to be overflowing with providers as the frontend continues to scale. 

With these small changes we can better support our developer's experience and drive a better development experience. 

### Changes 🏗️

- Addition of a basic theme from shadCn as mentioned here in there docs: [ShadCN Theming Documentation](https://ui.shadcn.com/docs/theming)
- Addition of tailwind configuration's for shadCn's theming class names. 
- Addition of `providers.tsx` this file will house the providers leveraged by the frontend solution. 
- Addition but commented until further confirmation of setting the theme based off of the system's preference. This could be a good call for a user preference setting in our settings dropdown. 
```
      <Providers
          attribute="class"
          defaultTheme="light"
          // Feel free to remove this line if you want to use the system theme by default
          // enableSystem
          disableTransitionOnChange
      >
```

Resolves : [Standardize Frontend Theme](https://github.com/Significant-Gravitas/AutoGPT/issues/7624) #7624 
- 
### PR Quality Scorecard ✨

- [x] Have you used the PR description template? &ensp; `+2 pts`
- [x] Is your pull request atomic, focusing on a single change? &ensp; `+5 pts`
- [x] Have you linked the GitHub issue(s) that this PR addresses? &ensp; `+5 pts`
- [x] Have you documented your changes clearly and comprehensively? &ensp; `+5 pts`
- [ ] Have you changed or added a feature? &ensp; `-4 pts`
  - [ ] Have you added/updated corresponding documentation? &ensp; `+4 pts`
  - [ ] Have you added/updated corresponding integration tests? &ensp; `+5 pts`
- [ ] Have you changed the behavior of AutoGPT? &ensp; `-5 pts`
  - [ ] Have you also run `agbenchmark` to verify that these changes do not regress performance? &ensp; `+10 pts`

___

### **PR Type**
Enhancement, Documentation

___

### **Description**
- Centralized theme and tooltip providers into a new `Providers` component to simplify the root layout.
- Updated Tailwind configuration to support new theme settings, including colors and fonts.
- Defined CSS variables for light and dark themes and applied them to base styles.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>layout.tsx</strong><dd><code>Centralize theme and tooltip providers in layout</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rnd/autogpt_builder/src/app/layout.tsx

<li>Removed individual theme provider in favor of centralized <code>Providers</code> <br>component.<br> <li> Simplified imports by removing unused icons.<br> <li> Updated <code>RootLayout</code> to use the new <code>Providers</code> component.<br>

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7622/files#diff-7c8736b609b98c456d0903f1b1f50296e2143f00d6530475522bc96d62481097">+43/-46</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>providers.tsx</strong><dd><code>Create centralized Providers component</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rnd/autogpt_builder/src/app/providers.tsx

<li>Added new <code>Providers</code> component to centralize theme and tooltip <br>providers.<br>

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7622/files#diff-04a63f04b59a0a6bc3538188241289f779ce2885023746f6a1230cfd1928e212">+14/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>tailwind.config.ts</strong><dd><code>Update Tailwind configuration for new theme settings</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rnd/autogpt_builder/tailwind.config.ts

<li>Updated Tailwind configuration to include new theme colors and font <br>settings.<br> <li> Added new content paths for Tailwind to scan.<br>

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7622/files#diff-7b06c9faf80ca70e561949e468f496bccb6f7a2528a153f2e359511f1975ea73">+58/-12</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>globals.css</strong><dd><code>Define and apply CSS variables for theming</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rnd/autogpt_builder/src/app/globals.css

<li>Added CSS variables for light and dark themes.<br> <li> Applied base styles using new theme variables.<br>

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7622/files#diff-f831c19bcd908988823cf77817854d3467f6b1c1d1cee65a51105c0b01c3bf82">+57/-0</a>&nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

**Comment by andrewhooker2** [comment]
While the above suggestions from the bot are viable they are not relvant to the PR changes that were made

**Comment by ntindle** [comment]
> While the above suggestions from the bot are viable they are not relvant to the PR changes that were made

That happens sometimes. Its often useful when someone messes up something from a security perspective lol

File: rnd/autogpt_builder/src/app/layout.tsx [hunk 1]
```diff
@@ -1,10 +1,8 @@
 import React from 'react';
 import type { Metadata } from "next";
-import { ThemeProvider as NextThemeProvider } from "next-themes";
-import { type ThemeProviderProps } from "next-themes/dist/types";
 import { Inter } from "next/font/google";
```
Comment (Pwuts) [inline rnd/autogpt_builder/src/app/layout.tsx:3]: bad indentation in this file

File: rnd/autogpt_builder/tailwind.config.ts [hunk 2]
```diff
@@ -3,7 +3,9 @@ import type { Config } from "tailwindcss";
 const config = {
   darkMode: ["class"],
   content: [
-    './src/**/*.{ts,tsx}',
+    './components/**/*.{ts,tsx}',
+    './app/**/*.{ts,tsx}',
+    './src/**/*.{ts,tsx}'
```
Comment (Pwuts) [inline rnd/autogpt_builder/tailwind.config.ts]: there's nothing outside of `src`

**Review by Pwuts — COMMENTED** [review]
I don't think we're in need of a different shad/cn theme: I actually chose shad/cn because I like its base style. The cause of current style inconsistencies is that we have a mix of native shad/cn components and custom styled ones. Current plan to fix that is to strip the custom styling and move to unaltered shad/cn components, like in #7622.

The only thing we'd like to add soon (theme-wise) is a working dark mode. It's not a high priority but definitely nice to have. This can hopefully be implemented with minimal changes, see the shad/cn docs for more info.

File: rnd/autogpt_builder/tailwind.config.ts (hunk 2 above)
Comment (andrewhooker2) [reply rnd/autogpt_builder/tailwind.config.ts]: Thank you for the call out on this I have reverted this change

File: rnd/autogpt_builder/src/app/layout.tsx (hunk 1 above)
Comment (andrewhooker2) [reply rnd/autogpt_builder/src/app/layout.tsx:3]: Oh thank you for noticing that. I've fixed the tab error on line 52. 

In a future PR would we like to standardize this with a prettier implementation?

Comment (ntindle) [reply rnd/autogpt_builder/src/app/layout.tsx:3]: Yeah that makes sense. Can you open an issue for it and link it here?

**Review by ntindle — APPROVED** [review]

**Review by ntindle — APPROVED** [review]

**Pull request merged**